sway/commands/pointer_constraint.c

```diff
@@ -0,0 +1,54 @@
+#include <string.h>
+#include <wlr/types/wlr_pointer_constraints_v1.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/input/cursor.h"
+#include "sway/input/seat.h"
+
+enum operation {
+	OP_ENABLE,
+	OP_DISABLE,
+	OP_ESCAPE,
+};
+
+// pointer_constraint [enable|disable|escape]
+struct cmd_results *cmd_pointer_constraint(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "pointer_constraint", EXPECTED_EQUAL_TO, 1))) {
+		return error;
+	}
+
+	enum operation op;
+	if (strcmp(argv[0], "enable") == 0) {
+		op = OP_ENABLE;
+	} else if (strcmp(argv[0], "disable") == 0) {
+		op = OP_DISABLE;
+	} else if (strcmp(argv[0], "escape") == 0) {
+		op = OP_ESCAPE;
+	} else {
+		return cmd_results_new(CMD_FAILURE, "Expected enable|disable|escape");
+	}
+
+	if (op == OP_ESCAPE && config->reading) {
+		return cmd_results_new(CMD_FAILURE, "Can only escape at runtime.");
+	}
+
+	struct sway_cursor *cursor = config->handler_context.seat->cursor;
+	struct seat_config *seat_config = seat_get_config(cursor->seat);
+	switch (op) {
+	case OP_ENABLE:
+		seat_config->allow_constrain = true;
+		break;
+	case OP_DISABLE:
+		seat_config->allow_constrain = false;
+		/* fallthrough */
+	case OP_ESCAPE:
+		if (cursor->active_constraint) {
+			wlr_pointer_constraint_v1_send_deactivated(
+					cursor->active_constraint);
```

Comment:
`cursor->active_constraint` needs to be unset too. It's probably better to call `sway_cursor_constrain(NULL)`.